kamon-core/src/main/scala/kamon/trace/SpanPropagation.scala

```diff
@@ -194,6 +199,80 @@ object SpanPropagation {
   }
 
 
+  /**
+   * Reads and Writes a Span instance using the jaeger single-header propagation format.
+   * The specification and semantics can be found here:
+   *   https://www.jaegertracing.io/docs/1.7/client-libraries/#propagation-format
+   *
+   * The description somewhat ambiguous, a lots of implementation details are second-guessed from existing clients
+   */
+  object UberSpanCodec {
+    def apply(): UberSpanCodec = new UberSpanCodec()
+    val HeaderName = "uber-trace-id"
+    val Separator = ":"
+    val Default = "0"
+    val DebugFlag = "d"
+  }
+
+  class UberSpanCodec extends Propagation.EntryReader[HeaderReader] with Propagation.EntryWriter[HeaderWriter] {
+    import UberSpanCodec._
+
+    override def write(context: Context, writer: HttpPropagation.HeaderWriter): Unit = {
+      val span = context.get(Span.Key)
+
+      if (span != Span.Empty) {
+        val parentContext = if (span.parentId != Identifier.Empty) span.parentId.string else Default
+        val sampling = encodeSamplingDecision(span.trace.samplingDecision)
+        val debug: Byte = 0
+        val flags = (sampling + (debug << 1)).toHexString
+        val headerValue = Seq(span.trace.id.string, span.id.string, parentContext, flags).mkString(Separator)
+
+        writer.write(HeaderName, urlEncode(headerValue))
```

Comment:
Why is the value being url-encoded? I couldn't find anything in the Jaeger docs saying that it should be like that :confused:

Reply:
The `TextMapCodec` on jaeger side can be configured to both flavors (`urlEncoded` or not). The urlEncode reflects the default configuration. To formulate it differently, it didn't worked for us out-of-the box otherwise.